app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt

```diff
@@ -0,0 +1,155 @@
+
+    private val _navigationListener = SingleLiveEvent<Int>()
+    val navigationListener = _navigationListener.asLiveData()
+
+    private val _addDoctorResponse = SingleLiveEvent<String>()
+    val addDoctorResponse = _addDoctorResponse.asLiveData()
+
+    private lateinit var fireStore: FirebaseFirestore
+    private lateinit var firebaseUser: FirebaseUser
+
+    init {
+        auth = FirebaseAuth.getInstance()
+        fireStore = FirebaseFirestore.getInstance()
+    }
+
+    private fun isAllValidate() {
+        isDataValid.value = (!doctorName.value.isNullOrEmpty() && !doctorEmail.value.isNullOrEmpty()
+                && !doctorContactNumber.value.isNullOrEmpty() && doctorNameError.value.isNullOrEmpty()
+                && doctorEmailError.value.isNullOrEmpty() && doctorContactNumberError.value.isNullOrEmpty())
+    }
+
+    fun isValidateName(text: CharSequence?) {
+        if (text?.toString().isNullOrEmpty() || ((text?.toString()?.length ?: 0) < 3)) {
+            doctorNameError.value = resourceProvider.getString(R.string.valid_name_desc)
+        } else if (text?.get(0)?.isLetter() != true) {
+            doctorNameError.value = resourceProvider.getString(R.string.valid_name_start_with_char)
+        } else {
+            doctorNameError.value = null
+        }
+        isAllValidate()
+    }
+
+    fun isValidEmail(text: CharSequence?) {
+        if (text?.toString().isNullOrEmpty() || text?.toString()?.isEmailAddressValid()
+                ?.not() == true
+        ) {
+            doctorEmailError.value = resourceProvider.getString(R.string.enter_valid_email)
+        } else {
+            doctorEmailError.value = null
+        }
+        isAllValidate()
+    }
+
+    fun isValidContact(text: CharSequence?) {
+        if (text?.toString().isNullOrEmpty()) {
+            doctorContactNumberError.value =
+                resourceProvider.getString(R.string.error_valid_phone_number)
+        } else {
+            if (Patterns.PHONE.matcher(text ?: "").matches()) {
+                if (text?.toString()?.length == 10) {
+                    doctorContactNumberError.value = null
+                } else {
+                    doctorContactNumberError.value =
+                        resourceProvider.getString(R.string.error_valid_phone_number)
+                }
+            } else {
+                doctorContactNumberError.value =
+                    resourceProvider.getString(R.string.error_valid_phone_number)
+            }
+        }
+        isAllValidate()
+    }
+
+
+    fun addDoctorData() {
+
+        firebaseUser = auth?.currentUser!!
+        if (firebaseUser != null) {
+            // when firebaseUser is not null then
+            val userData = UserDataRequestModel(
+                userId = firebaseUser.uid,
+                isDoctor = true,
+                email = doctorEmail.value,
+                name = doctorName.value,
+                contactNumber = doctorContactNumber.value,
```

Comment:
Here only pass form data that is show in UI. Still passing other data. Put default value on data class.

not define each and every field default value null.